R/plotting.R

```diff
@@ -220,7 +220,8 @@ plotBinScatter <- function(x, y, b,
 #'     (default: 99.5th percentile).
 #' @param highlight A logical vector indicating motifs to be highlighted.
 #' @param cluster If \code{TRUE}, the order of transcription factors will be determined by
-#'     hierarchical clustering of the \code{"pearsonResid"} component. Alternatively, an
+#'     hierarchical clustering of the \code{"pearsonResid"} component. If there are NA
+#'     values, they will be replaced by zero to do the clustering. Alternatively, an
```

Comment:
I guess your idea to set these to zero was to keep the original behavior?
It seems that `dist()` can handle `NA` values in the vectors, so I think there is no need to set them to zero (that implies again that there is no enrichment, which we don't know). I think the default behaviour of `dist()` is preferrable, from the documentation:
 ```Missing values are allowed, and are excluded from all computations involving the rows within which they occur.```

Reply:
I ran into problems with the plotMorifHeatmap() function otherwise. I just tried to re-create it, and while the dist() function handles NAs, the problems arise with hclust(dist). It's just a problem if `cluster=TRUE` and there are NA values. I guess another option could be to discard those motifs altogether from the Heatmap. Unless, there could be cases where NAs are in some bins but not all (then removing the entire motif isn't ideal), which could happen if background="genome" for example.

Reply:
I agree that leaving out these motifs would not be a good idea, e.g. if the `NA` is only occurring in some of the bins.
I think the problems in `plotMotifHeatmaps(..., cluster = TRUE)` may be unrelated to `hclust(dist(...))`, but may come from `quantile(...)` that is used to calculate the range of values for color mapping.
I have tried to add `na.rm = TRUE` to these calls to see if that resolves it (commit f49afeb). Could you run your tests again and check if that resolves the issue?

Reply:
ah yes good point, I missed that. I ran an example where one motif had NA values, and once I enter the plotMotifHeatmaps() function, if we don't deal with NA values, hclust throws this error: 
`Error in hclust(dist(pearsResidAssay)) : 
  NA/NaN/Inf in foreign function call (arg 10)`

I ran the example with the latest commit and there are no errors with the plot function :)

Reply:
You are right, there is still a problem for motifs that are completely `NA` in all bins. I made another commit (a9c43e0), which checks for this case and removes such motifs with a warning. Would that be a good alternative? This way, we neither claim no enrichment (by setting `NA`s to zero), nor do we miss motifs that have `NA`s in only a subset of bins.

Reply:
yes, I think this is a better solution. I've tried it on the example where all values for a motif were NA and it works nicely. I was thinking if we want to show in the legend the color for NA values, but I couldn't find an option in Heatmap() that adds a legend for `na_col`, so I think we can leave it like this. 
As for testing, since we read in the `rds` file of the enrichment result, we could test `plotMotifHeatmaps()` on instances where a motif has NAs in all bins, but we would need to update the saved `rds` file (it's based on the previous implementations and has no NAs). Shall I do that and add a test covering the NA instance?

Reply:
I have added a test in which I use the saved `rds` and just modify the `pearsonResid` assay to have all-NA-motifs. I also adjusted the enrichment calculation tests that were now missing some parts due to the new default test (`"fisher"` instead of `"binom"`). This is in ffd53d9. If you are happy with this, I think we can merge the PR.

Reply:
perfect, thank you for your feedback and fixes! I also think we can merge the PR.